scavenger-api/src/main/kotlin/com/navercorp/scavenger/service/SnapshotNodeService.kt

```diff
@@ -112,40 +113,40 @@ class SnapshotNodeService(
     * "a.b.c.d(e, f)" to ["a", "b", "c", "d(e, f)"]
     * "a.b.c.D(e, f)" to ["a", "b", "c", "D", "D(e, f)"]
     */
-    private fun splitSignature(methodInvocationEntity: MethodInvocationEntity): List<String> {
+    private fun splitSignature(methodInvocationEntity: MethodInvocationEntity): List<Map<String, String>> {
         val (nameOnlySignature, arguments) = methodInvocationEntity.signature.split("(", limit = 2)
 
-        val elements = nameOnlySignature.split(*DELIMITERS)
+        val elements = nameOnlySignature.split("(?=\\b[.$])".toRegex())
             .filterNot { it == "" }
+            .map { mutableMapOf("signature" to it.replace(".", "")) }
             .toMutableList()
 
         if (isConstructor(methodInvocationEntity)) {
-            elements.add(elements.last())
+            val constructorSignature = checkNotNull(elements.last()["signature"]) { "signature must be not null" }
+            elements.add(mutableMapOf("signature" to constructorSignature.replace("$", "")))
+        }
+
+        val lastElement = elements.last()
+        lastElement["signature"] = "${lastElement["signature"]}($arguments" + if (arguments.last() != ')') ")" else ""
```

Comment:
Is any case other than `)` at the end of the argument?

Reply:
maybe as follow case.
if end like this ...(truncated), we add `)` manually.